### ISSUE TITLE
Some monitors have full brightness at ~40% slider

### ISSUE THREAD

**Issue by willerbee** [the0neyouseek/MonitorControl#207]
I would think that this is because my monitor probably didn't implement DDC/CI well (it's an AN-238W03K). But I can control brightness and volume alright. This gem is such a blessing. 

Volume slider controls monitor volume well, but with brightness slider at roughly 40% in the app my monitor is at 94% brightness. From there, pressing "brightness up" and my monitor goes blank (because the monitor is being set to something equivalent to 100+). Doing another "brightness up" keypress and the monitor works again but reports at 142% brightness 😄 and it's a little bit brighter (and worrysome it could break lol). It would be nice to be able to have some kind of scale/override/calibration to set in the preferences so that I can let the app know that 40% slider is the real 100% brightness.

**Comment by kumowoon1025** [comment]
Yeah, display controllers can literally be on a spectrum of "speaks perfect DDC/CI MCCS VCP" to "hi u wont pixels mate?" even on brand name models. 

In particular they can be pretty horrible at reporting luminance state & capability, especially if they support NCL curves with HDR. The display itself reporting 100%+ brightness is weird though, is it supposed to do that? (extended range or something?)

**Comment by willerbee** [comment]
LOL. I don't think it is supposed to do that because if you control the brightness using the monitor's controls at the back, 100% is the max you can go. They might have restricted past 100% on hardware buttons and forgot about limiting it through DDC/CI. This monitor supports HDR though (as Windows 10 reports), but it feels like emulated.

**Comment by kumowoon1025** [comment]
Windows 10 reports any display capable of 8bit+FRC as HDR  capable I think (Windows HD Wide Color or something). Maybe that's why it feels unnatural... At the same time though, HDR displays are supposed to be able to represent brighter-than-100% brightness for a short time or in a small area (specular highlights, like maybe sunlight glinting off in an already lit up scene)

I've never even heard of _brightness_ going over 100% though. Volume maybe, if they have European max limits set or have a preamp or something.Did you try seeing if macOS detects it as an HDR display?

**Comment by willerbee** [comment]
I don't know about this AN-238W03K if it is 8bit+FRC, but I had a Dell P2415Q and it doesn't show Windows HD Color options like this one does. MacOS detects it like a regular 75Hz monitor.

Photos I took with OSD HDR setting, Youtube showing HDR quality options, and brightness setting indicating 94% and 142%
https://photos.app.goo.gl/xwvLaHjLfAi7JEdH8

I mentioned earlier that it feels like emulated HDR because Windows UI look like crap, but HDR videos (at least from youtube) look amazing.

**Comment by kumowoon1025** [comment]
Hey, don't know if this is related, but for what it's worth, I finally picked up my Thunderbolt Display that's been sitting in the closed Apple Store, (it's the original one from Apple, with TB1 and FW800/USB2.0 combo expansion on the back) I plugged it into an iMac with a missing display, and it did something similar.

100% brightness on MonitorControl, and the brightness HUD showed at 100% too (which is to be expected since it's overridden I guess). Quit the app, and used the driver or whatever macOS uses to control the display, and it went back to show it was at about ~20% brightness, which was how dim it seemed to me. I could then use the same brightness controls on my keyboard to bring it back up to full brightness.

Now that I recount it, maybe it's kind of the opposite of the issue you had, but I think a conflict between two control channels trying to light up or dim down the display could have been an issue. OP only used the app for volume control right?
